### PR TITLE
Pull request for airplane program

### PR DESCRIPTION
**Pull request by mikey-roberts**

### CODE REVIEW

**Comment by mikey-roberts** [comment]
- Would like to improve RSpec tests at some stage. 
- Missed test for the BONUS challenge to have multiple planes take off and return. 
- Met user stories

File: README.md [hunk 1]
```diff
@@ -60,30 +49,40 @@ To ensure safety
...
-**BONUS**
-
-* Write an RSpec **feature** test that lands and takes off a number of planes
-
-Note that is a practice 'tech test' of the kinds that employers use to screen developer applicants.  More detailed submission requirements/guidelines are in [CONTRIBUTING.md](CONTRIBUTING.md)
-
-Finally, don’t overcomplicate things. This task isn’t as hard as it may seem at first.
-
-* **Submit a pull request early.**
+What I've learnt -
```
Comment (SeanEmmers) [inline README.md:79]: Clear & concise README that makes it easy to understand the approach the developer.

File: lib/airport.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  end
+
+  def land(plane)
+    fail 'Unable to land. Weather is stormy' if stormy?
+    fail 'Airport is full' if airport_full?
+
+    planes << plane
+  end
+
+  def take_off
```
Comment (SeanEmmers) [inline lib/airport.rb:20]: Methods are clearly written and follow SRP very well.

File: lib/airport.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
+require_relative 'plane'
+
+class Airport
+  DEFAULT_CAPACITY = 5
+  attr_accessor :capacity, :planes
```
Comment (SeanEmmers) [inline lib/airport.rb:5]: Does the 'planes' variable need to have read & write access?

File: lib/airport.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  def airport_full?
+    planes.count == @capacity
+  end
+
+  def airport_empty?
+    planes.empty?
+  end
+
+  def stormy?
```
Comment (SeanEmmers) [inline lib/airport.rb:38]: Should the 'stormy?' method be inside the Airport class? Maybe there should be a class for weather

File: lib/plane.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    @flying = true
+  end
+  
+  def take_off
+    fail 'Plane is already in-flight' if @flying
+
+    in_flight
+  end
+
+  def land
```
Comment (SeanEmmers) [inline lib/plane.rb:14]: Great detail to avoid edge cases

File: spec/airport_spec.rb [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      before do 
+        allow(subject).to receive(:stormy?) { false }
+      end
+      it 'Plane landing when in the airport' do 
+        subject.land(plane)
+        expect(subject.planes).to eq [plane]
+      end
+    end
+
+    context 'when stormy' do
```
Comment (SeanEmmers) [inline spec/airport_spec.rb:31]: Good use of RSpec syntax I was unaware of

File: spec/plane_spec.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  it 'Error when trying to land when plane in-flight' do
+    expect { subject.take_off }.to raise_error 'Plane is already in-flight'
+  end 
+    
+  it 'Plane in-flight' do
+    subject.land
+    expect(subject.take_off).to eq true
+  end  
+  
+  it 'Error when trying to land when plane is already on the ground' do
```
Comment (SeanEmmers) [inline spec/plane_spec.rb:20]: All tests are clear and easy to understand, very good.

**Pull request closed without merging**